src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/PackagePropertyPage/PackageIconValueProvider.cs

```diff
@@ -0,0 +1,101 @@
+                ? string.Empty : (packagePath ?? string.Empty);
+            // The assumption is that PackagePath does not contain a path to a file; only a directory path.
+            return Path.Combine(packagePath, filename);
+        }
+
+        private async Task<IProjectItem?> GetExistingNoneItemAsync(string existingPackageIcon)
+        {
+            foreach (IProjectItem noneItem in await _sourceItemsProvider.GetItemsAsync(NoneItem))
+            {
+                string pack = await noneItem.Metadata.GetEvaluatedPropertyValueAsync(PackPropertyName);
+                string itemIconFilename = Path.GetFileName(noneItem.EvaluatedInclude);
+                string existingIconFilename = Path.GetFileName(existingPackageIcon);
+                // Instead of doing pure equality between itemPackageIcon and existingPackageIcon, a user may update the PackagePath
+                // of the item and forget to update the PackageIcon to reflect those changes, or vice versa. Instead, if the filename
+                // of this packed None item and the filename of the PackageIcon match, consider those to be related to one another.
+                if (TrueValue.Equals(pack, StringComparison.OrdinalIgnoreCase) && itemIconFilename.Equals(existingIconFilename, StringComparison.Ordinal))
+                {
+                    return noneItem;
+                }
+            }
+
+            return null;
+        }
+
+        public override async Task<string?> OnSetPropertyValueAsync(string propertyName, string unevaluatedPropertyValue, IProjectProperties defaultProperties, IReadOnlyDictionary<string, string>? dimensionalConditions = null)
+        {
+            string relativePath = PathHelper.MakeRelative(_unconfiguredProject, unevaluatedPropertyValue);
+            string existingPackageIcon = await defaultProperties.GetEvaluatedPropertyValueAsync(PackageIconPropertyName);
+            IProjectItem? existingItem = await GetExistingNoneItemAsync(existingPackageIcon);
+            string packagePath = string.Empty;
+            if (existingItem != null)
+            {
+                string itemPackagePath = await existingItem.Metadata.GetEvaluatedPropertyValueAsync(PackagePathPropertyName);
+                string itemPackageIcon = CreatePackageIcon(existingItem.EvaluatedInclude, itemPackagePath);
+                // The new filepath is the same as the current. No item changes are required.
+                if (relativePath.Equals(existingItem.EvaluatedInclude, StringComparison.Ordinal))
+                {
+                    return itemPackageIcon;
+                }
+
+                await existingItem.SetUnevaluatedIncludeAsync(relativePath);
+                packagePath = itemPackagePath;
+            }
+            else
+            {
+                await _sourceItemsProvider.AddAsync(NoneItem, relativePath, new Dictionary<string, string> { { PackPropertyName, "True" }, { PackagePathPropertyName, packagePath } });
```

Comment:
`TrueValue` instead of `"True"`?

Reply:
Fixed. I didn't have the constant at first and forgot to update usage of it here.